app/forms/concerns/piv_cac_form_helpers.rb

```diff
@@ -18,9 +18,9 @@ def token_decoded
 
   def not_error_token
     possible_error = @data['error']
+    self.key_id = @data['key_id']
```

Comment:
If we're wanting `key_id` to be assigned in all cases, should we consider setting this at the same time we're assigning `@data` in `token_decoded` above? Rather than trying to capture it in each of these error handling scenarios between `not_token_error` and `token_has_correct_nonce`.

Reply:
good idea.